api/src/play_asset_pack_request_tracker.gd

```diff
@@ -0,0 +1,65 @@
+# ##############################################################################
+#
+#	Copyright 2020 Google LLC
+#
+#	Licensed under the Apache License, Version 2.0 (the "License");
+#	you may not use this file except in compliance with the License.
+#	You may obtain a copy of the License at
+#
+#		https://www.apache.org/licenses/LICENSE-2.0
+#
+#	Unless required by applicable law or agreed to in writing, software
+#	distributed under the License is distributed on an "AS IS" BASIS,
+#	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#	See the License for the specific language governing permissions and
+#	limitations under the License.
+#
+# ##############################################################################
+#
+# The PlayAssetPackRequestTracker class generates unique signal_id integers and
+# keeps a mapping of signal_id to Request objects. This signal_id is used as an
+# identifier and passed to the Android plugin. Eventually the Android plugin will
+# emit signals along with this signal_id. In this way we can know which signal 
+# emitted from the plugin corresponds to which Request object
+#
+# ##############################################################################
+class_name PlayAssetPackRequestTracker
+extends Object
+
+const _SIGNAL_ID_MIN = 0
+
+var _signal_id_counter : int
+var _register_request_mutex : Mutex
+var _signal_id_to_request_map : Dictionary
+
+func _init():
+	_signal_id_counter = _SIGNAL_ID_MIN
+	_signal_id_to_request_map = Dictionary()
+	_register_request_mutex = Mutex.new()
+
+func get_current_signal_id() -> int:
+	return _signal_id_counter
+
+func _increment_signal_id() -> void:
```

Comment:
Since this is now a very simple method and it's only used in one place, it no longer needs to be it's own method. Feel free to inline `_signal_id_counter += 1` in `register_request`.

Reply:
Refactored this line out of it's own method.